components/FileTable.tsx

```diff
@@ -12,15 +12,17 @@ import { faDownload, faSearch } from '@fortawesome/free-solid-svg-icons';
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
 
 interface IFileDownloadModalProps {
-    filenames: string[];
+    files: Entity[];
     onClose: () => void;
     isOpen: boolean;
 }
 
 const FileDownloadModal: React.FunctionComponent<IFileDownloadModalProps> = (
     props
 ) => {
-    const script = props.filenames.map((f) => `synapse get ${f}`).join('\n');
+    const script = props.files
+        .map((f) => `synapse get ${f.HTANDataFileID}`)
```

Comment:
@inodb can you confirm that `HTANDataFileID` is actually the synapse id?

Reply:
Hmm, I don't think `HTANDataFileID` is the right id. Synapse ids should start with `syn`. Actually we have a field called `synapseId` which is currently missing in our `Entity` model: https://github.com/ncihtan/htan-portal/blob/b695603fb1f5622e74245da6db66732bbfbefca6/lib/helpers.ts#L71

But this synapse id is incorrect for most of the files. There are a lot of `undefined`s `null`s  and `2-Processed`s. Should we assume that this will be fixed at some point?

Reply:
@milen-sage do you happen to know why the synapse ids might be missing? Is that on purpose? I can dig further if you don't

Reply:
@milen-sage Not sure about exact number but perusing the JSON it's lot of the `bts:ImagingLevel2` from `HTA11`. Looks like this one doesn't have an `entityId` column:

https://www.synapse.org/#!Synapse:syn25127098